src/components/Navigation/components/Item/Item.tsx

```diff
@@ -263,6 +251,37 @@ export class BaseItem extends React.Component<CombinedProps, State> {
     );
   }
 
+  private getBadgeMarkup() {
+    const {
+      badge,
+      new: isNew,
+      polaris: {intl},
+    } = this.props;
+
+    if (badge == null && !isNew) {
+      return null;
+    }
+
+    let badgeMarkup: Props['badge'] = null;
+    if (isNew) {
+      badgeMarkup = intl.translate('Polaris.Badge.STATUS_LABELS.new');
+    } else if (typeof badge === 'string') {
+      badgeMarkup = badge;
```

Comment:
Seems weird to call this `badgeMarkup` when it is actually only the text contents?